src/highdicom/pr/content.py

```diff
@@ -737,3 +757,1295 @@ def __init__(
+
+            softcopy_voi_luts.append(
+                SoftcopyVOILUT(
+                    window_center=center,
+                    window_width=width,
+                    window_explanation=exp,
+                    voi_lut_function=func,
+                    referenced_images=refs_to_include
+                )
+            )
+
+        for lut_id, im_list in by_lut.items():
+            if len(im_list) == len(referenced_images):
+                # All datasets included, no need to include the referenced
+                # images explicitly
+                refs_to_include = None
+            else:
+                # Include specific references
+                refs_to_include = ReferencedImageSequence(im_list)
+
+            luts = [
+                LUT(
+                    first_mapped_value=fmv,
+                    lut_data=np.frombuffer(
+                        data,
+                        np.uint8 if ba == 8 else np.uint16
+                    ),
+                    lut_explanation=exp
+                )
+                for (fmv, ba, exp, data) in lut_id
+            ]
+            softcopy_voi_luts.append(
+                SoftcopyVOILUT(
+                    referenced_images=refs_to_include,
+                    luts=luts
+                )
+            )
+
+        dataset.SoftcopyVOILUTSequence = softcopy_voi_luts
+
+    return dataset
+
+
+def _extract_icc_profile_attributes(
+    referenced_images: Sequence[Dataset]
+) -> Dataset:
+    """Get attributes of module ICC Profile from a referenced image.
+
+    Parameters
+    ----------
+    referenced_images: Sequence[pydicom.Dataset]
+        Image datasets from which to extract an ICC profile
+
+    Returns
+    -------
+    pydicom.Dataset
+        Dataset containing attributes of module ICC Profile
+
+    Raises
+    ------
+    ValueError:
+        When no ICC profile is found in any of the referenced images or if
+        more than one unique profile is found.
+
+    """
+    icc_profiles = []
+    for im in referenced_images:
+        if hasattr(referenced_images, 'ICCProfile'):
+            icc_profiles.append(im.ICCProfile)
+        elif hasattr(im, 'OpticalPathSequence'):
+            if len(im.OpticalPathSequence) > 1:
+                raise ValueError(
+                    'Cannot extract ICC Profile from referenced image. '
+                    'Color image is expected to contain only a single optical '
+                    'path.'
+                )
+            icc_profiles.append(im.OpticalPathSequence[0].ICCProfile)
+
+    if len(icc_profiles) == 0:
+        raise ValueError(
+            'Could not find an ICC Profile in any of the referenced images.'
+        )
+    if len(set(icc_profiles)) > 1:
+        raise ValueError(
+            'Found more than one ICC Profile in referenced images.'
+        )
+
+    dataset = Dataset()
+    dataset.ICCProfile = icc_profiles[0]
+    return dataset
```

Comment:
Not sure I really understand the purpose of returning a dataset here, rather than just the bu
ytes?

Reply:
Yeah..you are probably right. I just thought if these "extract" functions return more than one value, then it may get messy returning a tuple and it may make more sense to bundle the extracted attributes in a dataset. Your call.

Reply:
I would go with returning bytes. This is private so no need to overthink it

Reply:
Addressed with 424cad25d733d1cca21c7997a42ddbd6732ff054